meson.build

```diff
@@ -0,0 +1,43 @@
+project('dwl', ['c', 'cpp'],
+  version : '0.4'
+)
+
+subdir('protocols')
+
+cc = meson.get_compiler('c')
+libm = cc.find_library('m')
+xcb = dependency('xcb', required : get_option('xwayland'))
+xlibs = dependency('xcb-icccm', required : get_option('xwayland'))
+wayland_server_dep = dependency('wayland-server')
+wlroots_dep = dependency('wlroots')
+xkbcommon_dep = dependency('xkbcommon')
+libinput_dep = dependency('libinput')
+
+c_args = [
+  '-DWLR_USE_UNSTABLE',
+  '-D_POSIX_C_SOURCE=200809L',
+  '-DVERSION="@0@"'.format(meson.project_version())
+]
+
+if xcb.found() and xlibs.found()
```

Comment:
I think `and get_option('xwayland')` is missing here - I build my dwl with Xwayland disabled although I have these optional dependencies installed.

Reply:
If the xwayland option is disabled and I put the `get_option` on the dependencies, meson will automatically resolve them as not found

Reply:
Indeed, Meson's feature options are specifically designed to guarantee that you can single-source your information as much as possible. Pass one to a dependency lookup, and if it's disabled then the dependency lookup will log "Dependency foobar not found: feature **xwayland** disabled". No need to add if/else logic to sometimes check for them and sometimes not. No need to gate other checks on both "was the dependency found" and "was the dependency configured to be used".

Reply:
In fact, `and get_option('xwayland')` will reduce to `and <feature_object>` which doesn't have a boolean state so that's just an error. You'd have to use `and <feature_object>.allowed()` to get a valid boolean check.